### ISSUE TITLE
CNN Output Formulas are Confusing

### ISSUE THREAD

**Issue by dsilver829** [udacity/sdc-issue-reports#566]
Term 1 Lesson 6 (Convolutional Neural Networks): I find it confusing that two different formulas for the output width after applying a filter are given in this lesson.  Initially, the formula from the tensorflow docs is used in Section 7.  Later the formula `(input_width - filter_width + 2 * P)/S + 1` is used in Section 10.  This latter formula is used in the Stanford cs231n course, with the convention that a combination of input width, filter width, strides and padding is not allowed if the formula does not give an integer.  This formula also assumes equal padding on the left and right hand sides.  Tensorflow does not follow these conventions.  The stanford formula is also confusing in the context of tensorflow because in tensorflow you cannot specify a padding parameter P, the only choices are SAME or VALID.  The concepts are confused again in the quiz in Section 10 where it says "Valid padding of size 1 (P)".  This doesn't make sense as Valid means no padding.

**Comment by awbrown90** [comment]
@dsilver829  got this one cleaned up in the lesson materials.
